### PR TITLE
Add way to turn off slit overlay if data is not NIRSpec

### LINKED ISSUE #433: Mosviz: have slit overlay plugin only be active/visible when using NIRSpec data
This ticket will track refactoring the `slit overlay` plugin once #394 is merged. The plugin is automatically active when using mosviz, although it should only be active when NIRSpec data is loaded.

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address the slit overlay plugin issuing a traceback anytime NIRISS data is loaded and different rows are selected.

Fixes #433 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py [hunk 1]
```diff
@@ -80,7 +80,11 @@ def place_slit_overlay(self, *args, **kwargs):
 
         # 'S_REGION' contains slit information. Bypass in case no images exist.
         if len(image_data) > 0:
-            if len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta:
+            # Only use S_REGION for Nirspec data, turn the plugin off
+            # if other data is loaded
+            if len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta \
+                    and 'INSTRUME' in spec2d_data[0].meta \
+                    and spec2d_data[0].meta['INSTRUME'].lower() == "nirspec":
```
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:87]: Usage of `\` line continuation is discouraged. Also, the logic can be a little shorter?

```suggestion
            # Only use S_REGION for Nirspec data, turn the plugin off
            # if other data is loaded
            if (len(spec2d_data) > 0 and 'S_REGION' in spec2d_data[0].meta
                    and spec2d_data[0].meta.get('INSTRUME', '').lower() == "nirspec"):
```

**Review by pllim — COMMENTED** [review]
The solution seems elegant, but how does one test this PR? Is there a dataset that would trigger the plugin and one that would hide it? I don't see any tests added.

**Comment by pllim** [comment]
Re: tests -- Oh, I just saw #769

**Comment by rosteen** [comment]
@pllim I took the liberty of committing your suggestion since @javerbukh is on vacation and I want to use this. I tested in Mosviz with some NIRISS data and it seems to work to disable the slit overlay, avoiding some error traces I was hitting earlier. Given that additional tests are in a follow-up PR, do you think we can merge this?

**Review by rosteen — APPROVED** [review]

**Review by pllim — APPROVED** [review]
Sure. #yolo

**Comment by rosteen** [comment]
Now if only Github wasn't broken so the tests would pass...

**Comment by pllim** [comment]
I am going to close/re-open to re-trigger CI.

**Pull request merged**